src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -2157,12 +2163,12 @@ SystemNative_SetSockOpt(intptr_t socket, int32_t socketOptionLevel, int32_t sock
         }
         else if (socketOptionName == PAL_SO_REUSEADDR)
         {
-            //
-            // On Windows, SO_REUSEADDR allows the address *and* port to be reused.  It's equivalent to 
-            // SO_REUSEADDR + SO_REUSEPORT other systems. 
-            //
+            // SO_REUSEPORT on Linux will do load-balancing accross multiple TCP sockets.
+            // This is different from Windows and BSD-Unix, so we won't set this for TCP.
+            // For UDP, we need to set SO_REUSEADDR and SO_REUSEPORT to support multicast use-cases
+            // where it is common to bind multiple sockets to the same address.
             int err = setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, optionValue, static_cast<socklen_t>(optionLen));
-            if (err == 0)
+            if (err == 0 && protocolType != PAL_PT_TCP)
```

Comment:
So if I do:
```C#
using System;
using System.Net.Sockets;

class Program
{
    static void Main()
    {
        var s = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
        Console.WriteLine(s.GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseAddress));
        s.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseAddress, 0);
        Console.WriteLine(s.GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseAddress));
    }
}
```
I'll get back `1 1` on Linux (as opposed to `0 0` on Windows), since on Linux with this change we're just ignoring it?

Reply:
On Linux we are using SO_REUSEADDR as a 'fake' toggle for TCP.
So this should return 0 0.
After a Bind, GetSocketOption(ReuseAddress) will return 1 always.